source/js/viewer-core.js

```diff
@@ -1350,16 +1351,18 @@ export default class ViewerCore
      *
      * @param pageIndex
      */
-    setCurrentPage (pageIndex)
+    setCurrentPages (activePage, visiblePages)
     {
-        if (this.viewerState.currentPageIndex !== pageIndex)
+        // these arrays will never contain objects so this should be safe
+        if (JSON.stringify(this.viewerState.currentPageIndices) !== JSON.stringify(visiblePages))
```

Comment:
Still not clear why you're using `stringify` here

Reply:
It's the shortest way I found to do an array equality test, and since the arrays only contain numbers it has no way to fail

Reply:
:( It smells like a hack, no?

Reply:
I don't think so, this is a safe way to compare arrays and a shorthand to the alternate approach (comparing each value in order, which is less elegant than this)

Reply:
You could do the same to anything (cast it to a string) and say a very similar thing. You're also negating some optimizations you could make; for example `a.len === b.len` would be a very easy and fast way of comparing the two as a first step; if they're not the same length, then they're not equal.

Reply:
I think you should humor me and implement a real array comparison. If you like, and if you're interested, you can try both (`JSON.stringify` and an array comparison algorithm) and wrap them in a `console.time` call and see which one is quicker.

https://developers.google.com/web/tools/chrome-devtools/console/console-reference

Reply:
sure